### PR TITLE
adding port_channel from puppetlabs netdev_stdlib

### PR DESCRIPTION
**Pull request by tphoney**
tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb passed in 101.50 seconds
      Test Suite: tests @ 2016-01-29 14:58:44 +0000

      - Host Configuration Summary -

              - Test Case Summary for suite 'tests' -
       Total Suite Time: 101.50 seconds
      Average Test Time: 101.50 seconds
              Attempted: 1
                 Passed: 1
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 1

      - Specific Test Case Status -

Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
Pending Tests Cases:

### CODE REVIEW

**Comment by tphoney** [comment]
@mikewiebe @chrisvanheuveln can you have a look at this. Hope all is well your side.

TP

**Comment by tphoney** [comment]
@glennmatthews forgot you. sorry.

File: examples/demo_netdev_port_channel.pp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Manifest to demo cisco_interface provider
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
```
Comment (glennmatthews) [inline examples/demo_netdev_port_channel.pp]: Nitpick - please double-check the copyright dates on these new files.

File: README.md [hunk 2]
```diff
@@ -1955,6 +1957,25 @@ Determines whether or not the config should be present on the device. Valid valu
...
 
+### Type: port_channel
+
+#### Parameters
+
+##### `ensure`
+Determines whether or not the config should be present on the device. Valid values are 'present' and 'absent'.
+
+##### `id`
+Channel group ID. eg 100  Valid value is an integer.
```
Comment (glennmatthews) [inline README.md]: Missing punctuation here? Same comment on `name` below.

File: tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+
+  case device
+  when /n7k/
+    id = 'default_properties'
+  when /n5k|n6k/
+    id = 'default_properties'
+  when /n3k|n9k/
+    id = 'default_properties'
+  end
```
Comment (glennmatthews) [inline tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb]: What's the purpose of the case statement here?

Comment (tphoney) [inline tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb]: I left this in for now, assuming that there will be further devices added for testing. I do not have access to them. following on from https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_interface_portchannel/test_interface_portchannel.rb

Comment (mikewiebe) [inline tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb]: We have a more streamlined way to handle this now.  Please use the `platform:` hash key instead.

https://github.com/cisco/cisco-network-puppet-module/blob/release_1.2.0/tests/beaker_tests/cisco_command_config/test_command_config.rb#L97

**Comment by tphoney** [comment]
@glennmatthews removed case statement.

**Comment by glennmatthews** [comment]
:+1:

**Pull request merged**